tests/scripts/all.sh

```diff
@@ -1024,6 +1024,43 @@ component_check_test_cases () {
     unset opt
 }
 
+component_check_test_dependencies () {
+    msg "Check: test case dependencies: legacy vs PSA" # < 1s
+
+    found="check-test-deps-found-$$"
+    expected="check-test-deps-expected-$$"
+
+    # Find legacy dependencies in PSA tests
+    grep 'depends_on' tests/suites/test_suite_psa* |
+        grep -Eo '!?MBEDTLS_[^: ]*' |
+        grep -v MBEDTLS_PSA_ |
+        sort -u > $found
+
+    # Expected ones with justification - keep in sorted order!
+    rm -f $expected
+    # No PSA equivalent - WANT_KEY_TYPE_AES means all sizes
+    echo "!MBEDTLS_AES_ONLY_128_BIT_KEY_LENGTH" >> $expected
+    # This is used by import_rsa_made_up() in test_suite_psa_crypto in order
+    # to build a fake RSA key of the wanted size based on
+    # PSA_VENDOR_RSA_MAX_KEY_BITS. The legacy module is only used by
+    # the test code and that's probably the most convenient way of achieving
+    # the test's goal.
+    echo "MBEDTLS_ASN1_WRITE_C" >> $expected
+    # No PSA equivalent - we should probably have one in the future.
+    echo "MBEDTLS_ECP_RESTARTABLE" >> $expected
+    # No PSA equivalent - needed by some init tests
+    echo "MBEDTLS_ENTROPY_NV_SEED" >> $expected
+    # Used by two tests that are about an extension to the PSA standard;
+    # as such, no PSA equivalent.
+    echo "MBEDTLS_PEM_PARSE_C" >> $expected
+
+    # Compare reality with expectation.
+    # We want an exact match, to ensure the above list remains up-to-date.
+    diff -U0 $expected $found
```

Comment:
If this fails on the CI, the output will be something like

```
******************************************************************
* check_test_dependencies: Check: test case dependencies: legacy vs PSA 
* Thu Nov  9 16:09:10 CET 2023
******************************************************************
--- check-test-deps-expected-1228447    2023-11-09 16:09:10.409940159 +0100
+++ check-test-deps-found-1228447       2023-11-09 16:09:10.409940159 +0100
@@ -4,0 +5 @@
+MBEDTLS_OOPS
```

Is this comprehensible enough for someone who's not deeply familiar with the problem we want to solve here?

It's annoying that this code doesn't remember where the forbidden identifiers were found. But admittedly that would be more work and I'm not sure it's justified.

Reply:
> Is this comprehensible enough for someone who's not deeply familiar with the problem we want to solve here?
> 
Not sure. I'll see what I can do to make it better.

> It's annoying that this code doesn't remember where the forbidden identifiers were found. But admittedly that would be more work and I'm not sure it's justified.

I'm not sure it's justified. Most of the time the test should pass, and when it fails finding an unexpected occurrence of `MBEDTLS_FOO` then it's easy enough for anyone investigating the problem to `grep -n MBEDTLS_FOO tests/suites/test_suite_psa*` - that's what I did for this PR anyway.

Reply:
> Not sure. I'll see what I can do to make it better.

I went with comments. I think if anyone sees output that doesn't make sense to them, looking at the component that generated this output is a reasonable next step.